### PR TITLE
Improve bulk quantiles

### PR DESCRIPTION
**Pull request by jturner314**
See jturner314/ndarray-stats#26.

### CODE REVIEW

File: src/sort.rs [hunk 1]
```diff
@@ -121,11 +122,12 @@ where
...
             let partition_index = self.partition_mut(pivot_index);
             if i < partition_index {
-                self.slice_mut(s![..partition_index]).get_from_sorted_mut(i)
+                self.slice_axis_mut(Axis(0), Slice::from(..partition_index))
+                    .get_from_sorted_mut(i)
             } else if i == partition_index {
                 self[i].clone()
             } else {
-                self.slice_mut(s![partition_index + 1..])
+                self.slice_axis_mut(Axis(0), Slice::from(partition_index + 1..))
```
Comment (LukeMathWalker) [inline src/sort.rs:130]: I can see the rationale for all the changes you have proposed, but I feel I am missing something here: why is `slice_axis_mut` preferable to `slice_mut`?

Comment (jturner314) [reply src/sort.rs:130]: It's just slightly lower overhead, that's all. `.slice_mut()` goes through all the n-dimensional slicing infrastructure and in the end [calls `.slice_axis_inplace()`](https://github.com/rust-ndarray/ndarray/blob/0.12.1/src/impl_methods.rs#L374); `.slice_axis_mut()` [jumps there directly](https://github.com/rust-ndarray/ndarray/blob/0.12.1/src/impl_methods.rs#L411). [Edit: In theory, the compiler should be able to eliminate the extra slicing infrastructure through constant propagation and inlining, but in practice, some overhead usually remains.] The performance difference is quite small in this case (~2% in the small array tests when I tried it earlier), so if `.slice_mut()` seems cleaner, that's fine too.

Comment (LukeMathWalker) [reply src/sort.rs]: I think

Comment (LukeMathWalker) [reply src/sort.rs:130]: Readibility-wise, I'd say they are the same. I was just curious to understand the rationale, your laser-focused optimization eye is something I am trying to learn from your PR reviews :stuck_out_tongue:

**Pull request merged**